designs/2024-baseline-support/README.md

```diff
@@ -0,0 +1,380 @@
+* Running `eslint ./src` again reports no violations but exits with a non-zero status code, indicating the suppressions file needs updating.
+
+To address this, a new option `--prune-suppressions` will be introduced to ESLint. This boolean flag removes resolved violations from the suppressions file without adding new ones, unlike `--suppress-all`.
+
+``` bash
+eslint --prune-suppressions ./src
+eslint --prune-suppressions --suppressions-location /home/user/project/mycache ./src
+```
+
+### Execution details
+
+The suggested solution always compares against the existing suppressions file, typically `.eslint-suppressions.json`, unless `--suppressions-location` is specified. This makes it easier for existing and new projects to adopt this feature without the need to adjust scripts in `package.json` and CI/CD workflows. 
+
+To perform the comparison, we will go through each result and message from `ESLint.lintFiles`, checking each error `(severity == 2)` against the suppressions file. By design, we ignore warnings since they don't cause eslint to exit with an error code and serve a different purpose. If the file and rule are listed in the suppressions file, we can move the message to `LintResult#suppressedMessages` and ignore the result message.
+
+Here is a high-level overview of the execution flow:
+
+1. **Check for Options**
+  * If both `--suppress-all` and `--suppress-rule` are passed, exit with an error (these options are mutually exclusive).
+  * If either option is passed, update the suppressions file based on the `results`.
+  * If no option is passed, check if the suppressions file exists, considering `--suppressions-location`.
+2. **Match Errors Against Suppressions**
+  * For each file, count the number of errors per rule.
+  * For each rule in each file, compare the number of errors against the counter from the suppressions file.
+  * If number of errors equals to the counter in the suppressions file, move the messages to `LintResult#suppressedMessages` and ignore the corresponding result messages. Remove the entry from the suppressions file, in memory.
+  * If the number of errors is less than the counter in the suppressions file, move the messages to `LintResult#suppressedMessages` and ignore the corresponding result messages. Also, set the counter to the new number, in memory.
+  * If the number of errors is greater than the counter in the suppressions file, report all the errors as usual. Remove the entry from the suppressions file, in memory.
+3. **Prune unmatched suppressions**
+  * If `--prune-suppressions` is passed, take the updated suppressions from memory to check which suppressions are left.
+  * For each suppression left, update the suppressions file by either reducing the count or removing the suppression.
+4. **Report and exit**
+  * Exit with a non-zero status if there are unmatched suppressions, optionally listing them in verbose mode.
```

Comment:
By verbose mode we probably mean when the [`--debug`](https://eslint.org/docs/latest/use/command-line-interface#--debug) flag is passed on the command line.